lib/http_url_validator.rb

```diff
@@ -0,0 +1,10 @@
+class HttpUrlValidator < ActiveModel::EachValidator
```

Comment:
I read about how the URI validation occurs in the Ruby gem(https://github.com/rubygems/rubygems/blob/master/lib/rubygems/specification_policy.rb#L450-L459)  and tried to mimic it since I figured we would want the same validations to happen in both places. 

I extracted this validation into a custom validator because I knew that other URLs would eventually be added to the user profile, and we could establish a consistent way of validating the URLs. 

I just also wanted to call out that although the matching pattern `validUriPattern = %r{\Ahttps?://([^\s:@]+:[^\s:@]*@)?[A-Za-z\d\-]+(\.[A-Za-z\d\-]+)+\.?(:\d{1,5})?([\/?]\S*)?\z} # :nodoc:` could have been achieved with the `validate_the_formatting_of`  a custom validation will allow us to add checks if we want to extract params such as `redirect`